src/basicio.cpp

```diff
@@ -1817,8 +1818,6 @@ namespace Exiv2 {
                     }
                 }
             }
-            blockIndex--;
-            blockSize = (long)p_->blocksMap_[blockIndex].getSize();
```

Comment:
Unfortunately, this code doesn't seem to be covered by any of our tests. But I am pretty sure that if this loop doesn't terminate early then, on the final iteration of the loop, this array access will attempt to access `blocksMap_[-1]`. Since all this code is doing is preloading `blockSize` for the next iteration, it is better to just load it at the beginning of the loop.